hack/reduce-snapshot.sh

```diff
@@ -29,9 +29,18 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-# verify if in json form
-SNAPSHOT="$(cat "${SNAPSHOT}" 2> /dev/null || echo "${SNAPSHOT}")"
-[[ ! "${SNAPSHOT}" =~ ^\s*\{ ]] && echo 'Error: Cannot load snapshot from JSON string or file' && exit 1
+
+# Make sure to move snapshot contents to the WORKING_SNAPSHOT location. Then allow jq to
+# work with it there. This avoids having to read SNAPSHOT to memory.
+
+if [[ -f "$SNAPSHOT" ]]; then
+  jq empty "$SNAPSHOT" || { echo "JSON is invalid"; exit 1; }
+  WORKING_SNAPSHOT="$SNAPSHOT"
+else
+  WORKING_SNAPSHOT="$(mktemp snapshot.XXXXXX)"
+  printf "%s" "$SNAPSHOT" > "$WORKING_SNAPSHOT"
+  jq empty "$WORKING_SNAPSHOT" || { echo "JSON is invalid"; exit 1; }
```

Comment:
Not a blocker, but: you could move this outside the if block and remove line 37.